File: OCP/ Kata_ Shops Types/SolucionJava/Shop/Shop/StandardShop.java
package SolucionJava.Shop.Shop;

public class StandardShop extends AbstractShop{
    private ShopTypeInterface shop_type;

    public StandardShop(String name, Integer phone, ShopTypeInterface shop_type) {
        this.shop_type = shop_type;

        super(name, phone);
    }

    public ShopTypeInterface getShopType() {
        return this.shop_type;
    }
}

File: OCP/ Kata_ Shops Types/SolucionPhp/Shop/Shop/StandardShop.php
<?php

declare(strict_types=1);

namespace SolucionPhp\Shop\Shop;

use SolucionPhp\Shop\Shop\Type\ShopTypeInterface;

class StandardShop extends AbstractShop
{
    private ShopTypeInterface $type;

    public function __construct(string $name, int $phone, ShopTypeInterface $type)
    {
        parent::__construct($name, $phone);

        $this->type = $type;
    }

    public function getShopType(): ShopTypeInterface
    {
        return $this->type;
    }
}
